src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.cs

```diff
@@ -1926,8 +1930,25 @@ private static TimeZoneInfoResult TryGetTimeZoneUsingId(string id, bool dstDisab
                     }
                     else
                     {
-                        value = new TimeZoneInfo(match._id, match._baseUtcOffset, match._displayName, match._standardDisplayName,
-                                              match._daylightDisplayName, match._adjustmentRules, disableDaylightSavingTime: false, match.HasIanaId);
+                        value = match;
```

Comment:
Do you need both value and match?  Seems like you could just make it `if (cachedData._systemTimeZones.TryGetValue(id, out value))` and then get rid of this else block?

Reply:
good idea! I have done the suggestion.